overlord/snapstate/snapmgr.go

```diff
@@ -209,6 +209,22 @@ func (snapst *SnapState) Block() []snap.Revision {
 
 var ErrNoCurrent = errors.New("snap has no current revision")
 
+// Retrieval functions
+var readInfo = readInfoImpl
+
+func readInfoImpl(name string, si *snap.SideInfo) (*snap.Info, error) {
+	info, err := snap.ReadInfo(name, si)
+	if _, ok := err.(*snap.NotFoundError); ok {
```

Comment:
Hm, that is a really nice idea. We could simply always return that the snap is in broken state. The downside might be that we want to get bugreports about the other errors (we don't know yet what those might be).